### ISSUE TITLE
Let us use our models.

### ISSUE THREAD

**Issue by Cytranics** [microsoft/vscode-copilot-release#67]
i see advanced options for keys, but they dont work. I want to use all 32k tokens i have ;)

**Comment by digitarald** [comment]
@Cytranics can you describe more about which models you'd want to use instead and the problems this solves – local, hosted, etc?

**Comment by Cytranics** [comment]
> @Cytranics can you describe more about which models you'd want to use instead and the problems this solves – local, hosted, etc?

Yeah I'd like to refactor code that takes up 18,000 tokens and currently I can't do that. So I would be willing to specify my observe. You are URL in token key to use copilot cuz I like the software. However, I can't send it more than 4000 tokens currently

**Comment by julian-shalaby** [comment]
[Here's](https://plugins.jetbrains.com/plugin/22895-chatwindow) an alternative which lets you use GPT4 with your own API key

**Comment by anthonykim1** [comment]
I think this is already done.

Press on +Add Models 
I believe this is already fixed, can you see if you can reproduce in the [Insiders build](https://code.visualstudio.com/insiders)? Let us know if not, latest copilot extension.

**Comment by Cytranics** [comment]
Thank you 3 years later ;)


[A close-up of a business card]



Book a meeting: https://calendly.com/cytranic/30-min-meeting-1



________________________________
